### PR TITLE
[OTE][Releases v0.4.0][XAI] Hot-fix for Detection fix two stage error

### PR DESCRIPTION
**Pull request by dongkwan-kim01**
1. Changed from default class-wise `DetSaliencyMapHook` in the `inferrer.py` in the MPA side to select the saliency map hook from the checking instance if it is `TwoStageDetector`.
2. Because the current CI test can't check the saliency map exporting because all the OTE side command is disabling the saliency map export, I've added a single-sample checking step in the `eval.py`, to check the saliency map export step.
-> This can be removed and also enhanced in the future if #1430 merged

### CODE REVIEW

**Comment by dongkwan-kim01** [comment]
run ote-test

File: ote_cli/ote_cli/tools/eval.py [hunk 1]
```diff
@@ -160,6 +162,13 @@ def main():
         InferenceParameters(is_evaluation=True),
     )
 
+    # saliency map export testing for the single data
+    test_sample = DatasetEntity(items=[validation_dataset[0]])
+    _ = task.infer(
```
Comment (sungmanc) [inline ote_cli/ote_cli/tools/eval.py]: This line made an error. please check it.
http://validationreports.sclab.intel.com:8006/reports/test_result/639731b97775e4fe8791b56e

image

**Review by sungmanc — CHANGES_REQUESTED** [review]
There were 2 FAIL cases on this CI test. One was related to OOM, however, the other is related to the current change. Please check it. I attached the figure.

File: ote_cli/ote_cli/tools/eval.py (hunk 1 above)
Comment (dongkwan-kim01) [reply ote_cli/ote_cli/tools/eval.py]: Thank you for your investigation. I've fixed it.

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Merged MPA PR. Could you update MPA commit?

**Comment by dongkwan-kim01** [comment]
> Merged MPA PR. Could you update MPA commit?

Thank you. I'll update it!

File: ote_cli/ote_cli/tools/eval.py [hunk 2]
```diff
@@ -160,6 +162,14 @@ def main():
...
     )
 
+    if len(validation_dataset) > 0:
+        # saliency map export testing for the single data
+        test_sample = DatasetEntity(items=[validation_dataset[0]])
+        _ = task.infer(
+            test_sample.with_empty_annotations(),
+            InferenceParameters(is_evaluation=False),
+        )
+
```
Comment (goodsong81) [inline ote_cli/ote_cli/tools/eval.py]: Could you explain this block?
This file is not part of test but a external CLI code.
I think it's not appropriate for us to insert this kind of test code in production code.
By the way, what is different from the code block above in line#159?

**Review by goodsong81 — CHANGES_REQUESTED** [review]

File: ote_cli/ote_cli/tools/eval.py (hunk 2 above)
Comment (goodsong81) [reply ote_cli/ote_cli/tools/eval.py]: Ah, I've missed the description. Will you remove this code block in the test enhancement PR?

Comment (dongkwan-kim01) [reply ote_cli/ote_cli/tools/eval.py]: Yes, but I agree it's not appropriate for the production code and it's not necessary so we can remove it in this PR.

**Comment by dongkwan-kim01** [comment]
run ote-test

File: ote_cli/ote_cli/tools/eval.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
 import json
 
 from ote_sdk.configuration.helper import create
+from ote_sdk.entities.datasets import DatasetEntity
```
Comment (goodsong81) [inline ote_cli/ote_cli/tools/eval.py]: Do we still need this?

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Where is the MPA commit change?

File: ote_cli/ote_cli/tools/eval.py (hunk 3 above)
Comment (dongkwan-kim01) [reply ote_cli/ote_cli/tools/eval.py]: Updated both MPA commit change and removing the ote side code change

**Comment by dongkwan-kim01** [comment]
run ote-test

**Review by goodsong81 — DISMISSED** [review]
LGTM, let's see the test result.

**Comment by chuneuny-emily** [comment]
Component test result: http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=pr-ote-test-837&environment=iotg-dev-workstation-10

File: third-party-programs.txt [hunk 4]
```diff
@@ -69,3 +69,78 @@ SOFTWARE.
...
    mail: tianhengcheng@gmail.com
    copyright@wondervictor
+
+
+-------------------------------------------------------------
+
+5. Python
+Python 3.0a1 License
+A. HISTORY OF THE SOFTWARE
+==========================
```
Comment (goodsong81) [inline third-party-programs.txt:79]: Is this relevant to this hot-fix?

Comment (goodsong81) [reply third-party-programs.txt:79]: Ah, found the request via chat. Never mind.

**Review by goodsong81 — APPROVED** [review]

**Review by chuneuny-emily — APPROVED** [review]
